### DISCUSSION TITLE
How project id is protected on Android SDK?

### DISCUSSION THREAD

**Discussion by Raenar4k** [WalletConnect/walletconnect-monorepo#3138] [Developer Support] [answered]
Hello. I am wondering how is project id protected from being used by 3rd party on Android SDK.
Usually you upload signature hash, so that only apps with listed signatures can use some token/id.
Here i don't see such thing.

So what is stopping someone from using our project id in their app?
As far as i understand now, android sample allows me to do same things with any project id.

**Comment by jakubuid** [comment, marked as answer]
Hello, when you create a project in https://cloud.walletconnect.com/ you are obliged to white list package names/bundle id that are allowed to use a given project id. The SDK sends the bundle id/package name as a part of the Relay request so it can be validated.

**Reply by Raenar4k** [reply]
I think at the time of the question `Jul 13, 2023 ` there was no such option.
Regardless, thanks for the answer!
Is there plans to use signature validation? As package can be changed by 3rd party to match original one, and tokens are easily leaked in apk decompilation.
